core/src/main/java/hudson/init/impl/InstallUncaughtExceptionHandler.java

```diff
@@ -70,10 +80,55 @@ public void uncaughtException(Thread t, Throwable ex) {
                        "A thread (" + t.getName() + '/' + t.getId()
                                      + ") died unexpectedly due to an uncaught exception, this may leave your Jenkins in a bad way and is usually indicative of a bug in the code.",
                        ex);
+
+            // If we have an exception, let's see if it's related with missing classes on Java 11 We reach
+            // here with a ClassNotFoundException in an action, for example.
+            reportMissingClassJava11Telemetry(ex);
         }
 
     }
 
     private InstallUncaughtExceptionHandler() {}
 
+    /**
+     * Report the class not found via {@link MissingClassTelemetry} if this exception is related.
+     * @param e the exception to look into
+     * @return true if this exception or a cause of this one or a suppressed exception of this one was reported.
+     * {@link #exceptionsToReportToJava11Telemetry}.
+     */
+    private static boolean reportMissingClassJava11Telemetry(@Nonnull Throwable e) {
```

Comment:
You're returning a `boolean` value that is not used. Any special reason?

Reply:
It's used on line 115, to stop the recursion through the suppressed exceptions

Reply:
right!. My bad